core/src/main/scala-2.12/cats/data/ChainCompanionCompat.scala

```diff
@@ -0,0 +1,42 @@
+package cats.data
+
+import cats.data.Chain.{nil, one, Wrap}
+import cats.kernel.compat.scalaVersionSpecific.IterableOnce
+
+import scala.collection.immutable.{Seq => ImSeq}
+import scala.collection.mutable.{Seq => MutSeq}
+
+private[data] trait ChainCompanionCompat {
+
+  /**
+   * Creates a Chain from the specified sequence.
+   */
+  def fromSeq[A](s: Seq[A]): Chain[A] =
+    s match {
+      case mut: MutSeq[A] => fromMutableSeq(mut)
+      case imm: ImSeq[A]  => fromImmutableSeq(imm)
+    }
```

Comment:
Not an objection but rather just a suggestion about the style.

I personally don't like such renames on imports, because they tend to introduce new not-well known names. Moreover, Scala Docs suggests a convention about it:
https://docs.scala-lang.org/overviews/collections/overview.html

I.e. you could import it in this way:
```scala
import scala.collection.immutable
import scala.collection.mutable
```
Then use it in the code:
```scala
s match {
  case seq: mutable.Seq[A] => fromMutableSeq(mut)
  case seq: immutable.Seq[A]  => fromImmutableSeq(imm)
}
```
That looks way clearer and easier to read, doesn't it?

Reply:
Yup, I very much feel the same way about this.